versioned_docs/version-0.2.0/connectors/io-crd-config/sink-crd-config.md

```diff
@@ -0,0 +1,159 @@
+| `customSerdeSources` | The map of input topics to SerDe class names (as a JSON string). |
+| `customSchemaSources` | The map of input topics to Schema class names (as a JSON string). |
+| `sourceSpecs` | The map of source specifications to consumer specifications. Consumer specifications include these options: <br />- `SchemaType`: the built-in schema type or custom schema class name to be used for messages fetched by the connector. <br />- `SerdeClassName`: the SerDe class to be used for messages fetched by the connector. <br />- `IsRegexPattern`: configure whether the input topic adopts a Regex pattern. <br />- `SchemaProperties`: the schema properties for messages fetched by the connector. <br />- `ConsumerProperties`: the consumer properties for messages fetched by the connector. <br />- `ReceiverQueueSize`: the size of the consumer receive queue. <br /> - `cryptoConfig`: cryptography configurations of the consumer. |
+
+## Resources
+
+When you specify a function or connector, you can optionally specify how much of each resource they need. The resources available to specify are CPU and memory (RAM).
+
+If the node where a Pod is running has enough of a resource available, it is possible (and allowed) for a Pod to use more resources than its `request` for that resource. However, a Pod is not allowed to use more than its resource `limit`.
+
+## Secrets
+
+Function Mesh provides the `secretsMap` field for Function, Source, and Sink in the CRD definition. You can refer to the created secrets under the same namespace and the controller can include those referred secrets. The secrets are provide by `EnvironmentBasedSecretsProvider`, which can be used by `context.getSecret()` in Pulsar functions and connectors.
+
+The `secretsMap` field is defined as a `Map` struct with `String` keys and `SecretReference` values. The key indicates the environment value in the container, and the `SecretReference` is defined as below.
+
+| Field | Description |
+| --- | --- |
+| `path` | The name of the secret in the Pod's namespace to select from. |
+| `key` | The key of the secret to select from. It must be a valid secret key.|
+
+Suppose that there is a Kubernetes Secret named `credential-secret` defined as below:
+
+```yaml
+apiVersion: v1
+data:
+  username: foo
+  password: bar
+kind: Secret
+metadata:
+  name: credential-secret
+type: Opaque
+```
+
+To use it in Pulsar Functions in a secure way, you can define the `secretsMap` in the Custom Resource:
+
+```yaml
+secretsMap:
+  username:
+    path: credential-secret
+    key: username
+  password:
+    path: credential-secret
+    key: password
+```
+
+Then, in the Pulsar Functions and Connectors, you can call `context.getSecret("username")` to get the secret value (`foo`).
+
+## Authentication
+
+Function Mesh provides the `tlsSecret` and `authSecret` fields for Function, Source and Sink in the CRD definition. You can configure TLS encryption and/or TLS authentication using the following configurations.
```

Comment:
```suggestion
Function Mesh provides the `tlsSecret` and `authSecret` fields for Function, Source, and Sink in the CRD definition. You can configure TLS encryption and/or TLS authentication using the following configurations.
```
Use commas to separate items in a series of three or more. Use a comma before the conjunction that precedes the final item. https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.b82f2ay5cpsc